apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -1002,10 +1008,11 @@ public async Task WhenSessionPoolReady(CancellationToken cancellationToken = def
             await _sessionPool.WhenPoolReady(sessionPoolSegmentKey, cancellationToken).ConfigureAwait(false);
         }
 
-        internal Task<PooledSession> AcquireSessionAsync(SpannerTransactionCreationOptions creationOptions, CancellationToken cancellationToken)
+        internal Task<PooledSession> AcquireSessionAsync(SpannerTransactionCreationOptions creationOptions, CancellationToken cancellationToken, out SpannerTransactionCreationOptions enhancedTransactionCreationOptions)
```

Comment:
Something like this:

```suggestion
        internal Task<PooledSession> AcquireSessionAsync(ref SpannerTransactionCreationOptions creationOptions, CancellationToken cancellationToken)
```

Reply:
If we make this a field passed by reference, we need to consider the calls from `EphemeralTransaction` and `RetriableTransaction` where the `_creationOptions` are private and readonly currently. This would mean we will have to remove the `readonly` modifier on these fields in these classes.

That seems fine to me since neither the `SpannerClientCreationOption` defaults nor the `SpannerTransactionCreationOptions` already set on the txn should change once objects for these types of transactions are created. 

But removing the `readonly` could potentially open up these fields to potential misuse and unintended modifications in the future. Thoughts?

Reply:
We discussed this offline, and we agreed to going back to using an `out` parameter and refactor once we have done multiplexed sessions.